bytes_ostream.hh

```diff
@@ -268,6 +268,35 @@ public:
         write(bytes_view(reinterpret_cast<const signed char*>(ptr), size));
     }
 
+    // Writes the fragmented view
+    template<FragmentedView View>
+    void write(View& v) {
+        // copy data into currently available space
+        while (current_space_left() > 0 && !v.empty()) {
+            auto this_size = std::min(v.current_fragment().size(), size_t(current_space_left()));
+            memcpy(_current->data + _current->frag_size, v.current_fragment().begin(), this_size);
+            _current->frag_size += this_size;
+            _size += this_size;
+            v.remove_prefix(this_size);
+        }
+
+        // copy rest of the fragments one by one into newly alloc'ed chunks
+        while (!v.empty()) {
+            auto this_size = std::min(v.current_fragment().size(), size_t(max_chunk_size()));
+            std::copy_n(v.current_fragment().begin(), this_size, alloc_new(this_size));
+            v.remove_prefix(this_size);
+        }
+    }
+
+    // Writes n bytes from the given fragmented view
+    template<FragmentedView View>
+    void write(View& v, size_t n) {
```

Comment:
This method and the one above mutate the View (i.e. remove a prefix from it).
This is evil because that's a different behavior than for `bytes_view`.
Calling `managed_bytes_view x; out.write(x)` will modify `x`.
Calling `bytes_view x; out.write(x)` won't modify `x`.
That's begging for trouble.

Also, in general, a call like `out.write(x)` doesn't usually modify `x`. So I think this interface is surprising, in a bad way.

Either rename this method to `write_and_remove_prefix` (and the one above to `write_and_make_empty` or something), or modify those methods (and their callers) so that they take `View`/`const View&`.

Reply:
I have removed the `write(View& v, size_t n)` entirely and changed `write(View& v)` to `write(View v)`. The callers now send a prefix to this method when required.